torch/onnx/_internal/diagnostics/infra/formatter.py

```diff
@@ -19,6 +20,35 @@
 ]
 
 
+class LazyString:
+    """A class to lazily evaluate a string.
```

Comment:
For my education, what does lazily evaluate a string mean?

Reply:
In short, it allows passing a lambda instead of string to `log` api. So logger, based on log level, decides whether to call the lambda. 

Useful when you have a message that takes a long time to format, you can log with debug level so normal export performance (default level info) is not affected.